### PR TITLE
Port admin Duplicates tab from v1 (#29)

### PR DESCRIPTION
**Pull request by alsiola**
## Summary

- Add duplicate member detection API with email-based and fuzzy name-based matching (using `nameSimilarity` from `@percy-main/shared`)
- Add merge preview endpoint showing both members' details, memberships, dependents, and charges side-by-side
- Add transactional merge endpoint that re-points foreign keys, preserves Stripe customer ID, and deletes the duplicate
- Port the Duplicates tab frontend from v1 with group cards, radio selection, merge preview modal, cross-email warning, and typed confirmation
- Wire DuplicatesTab into admin panel ACTIVE_TABS

### Key decisions
- Member fetches in `mergeMembers` happen inside the transaction to avoid TOCTOU race conditions
- `getMergePreview` uses explicit column selection (not `selectAll()`) to avoid leaking internal fields
- Email duplicate detection is retained in the API even though `member.email` has a unique constraint — guards against constraint relaxation or data migration edge cases
- PostgreSQL `COUNT(*)` aggregates use `sql<string>` + `Number()` per project convention

## Test plan

- [x] 3 unit tests (self-merge error, preview error cases)
- [x] 10 integration tests via testcontainers:
  - Fuzzy name-based duplicate detection
  - Correct response shape
  - Membership/dependent/charge count inclusion
  - Merge preview with cross-email detection
  - Merge FK re-pointing (membership, dependent, charge)
  - Stripe customer ID preservation (both directions)
  - 404 for non-existent members
- [x] typecheck, lint, format all pass
- [ ] Manual testing against local dev

Closes #29

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/api/src/features/admin/service.ts [hunk 1]
```diff
@@ -1178,6 +1185,330 @@ export function chasePayment(db: Kysely<DB>) {
...
+
+      if (!keepMember || !removeMember) {
+        const error = new Error(
+          "One or both member records not found",
+        ) as Error & { statusCode: number };
+        error.statusCode = 404;
+        throw error;
+      }
+
+      // Re-point all foreign keys from removeMember to keepMember
```
[the remark this answers is not in the document]
Comment (alsiola) [reply apps/api/src/features/admin/service.ts]: Fixed in 5c649f3 — added `matchday_player.member_id` to the FK repoint list inside the merge transaction. Good catch, I'd missed that table.
